test/GasAccounting.t.sol

```diff
@@ -99,307 +124,515 @@ contract MockGasAccounting is GasAccounting, Test {
-        mockGasAccounting.initializeLock{value: metacallValue}(executionEnvironment, gasMarker, 0);
-        uint256 rawClaims = (gasMarker + mockGasAccounting.FIXED_GAS_OFFSET()) * tx.gasprice;
-        initialClaims = rawClaims * (
-            mockGasAccounting.SCALE() + mockGasAccounting.ATLAS_SURCHARGE_RATE() + mockGasAccounting.BUNDLER_SURCHARGE_RATE()
-        ) / mockGasAccounting.SCALE();
+    function fundContract(uint256 amount) internal {
+        // Fund the contract with enough ETH
+        deal(address(mockGasAccounting), amount);
+        // Verify the contract has sufficient balance
+        assertEq(address(mockGasAccounting).balance, amount, "Contract should have enough ETH balance");
     }
 
-    function test_contribute() public {
-        // FIXME: fix before merging spearbit-reaudit branch
-        vm.skip(true);
-
-        vm.expectRevert(
-            abi.encodeWithSelector(AtlasErrors.InvalidExecutionEnvironment.selector, executionEnvironment)
-        );
+    function test_contribute_withInvalidExecutionEnvironment_revert() public {
+        // Expect revert when contribute is called without the proper setup
+        vm.expectRevert(abi.encodeWithSelector(AtlasErrors.InvalidExecutionEnvironment.selector, address(0)));
         mockGasAccounting.contribute();
 
+        tearDown();
+    }
+
+    function test_contribute() public {
+        // Set up the environment for a valid contribute call
         uint256 contributeValue = 1000;
         deal(executionEnvironment, contributeValue);
+        mockGasAccounting.setLock(executionEnvironment, 0, uint8(ExecutionPhase.SolverOperation));
 
+        // Perform the valid contribute call
         vm.prank(executionEnvironment);
         mockGasAccounting.contribute{ value: contributeValue }();
 
+        // Verify the balances after contribution
         assertEq(address(mockGasAccounting).balance, contributeValue);
         assertEq(mockGasAccounting.deposits(), contributeValue);
+
+        tearDown();
     }
 
-    function test_borrow() public {
-        // FIXME: fix before merging spearbit-reaudit branch
-        vm.skip(true);
+    function test_borrow_preOpsPhase() public {
+        uint256 borrowedAmount = 1e18;
 
-        uint256 borrowedAmount = 5000;
-
-        vm.expectRevert(
-            abi.encodeWithSelector(AtlasErrors.InvalidExecutionEnvironment.selector, executionEnvironment)
-        );
-        mockGasAccounting.borrow(borrowedAmount);
+        fundContract(borrowedAmount);
+        mockGasAccounting.setLock(executionEnvironment, 0, uint8(ExecutionPhase.PreOps));
 
         vm.prank(executionEnvironment);
-        vm.expectRevert(
-            abi.encodeWithSelector(AtlasErrors.InsufficientAtlETHBalance.selector, 0, borrowedAmount)
-        );
         mockGasAccounting.borrow(borrowedAmount);
 
-        deal(
-            address(mockGasAccounting),
-            initialClaims + borrowedAmount // claims + amount borrowed
+        assertEq(
```

Comment:
This looks good. Maybe also test in each borrow test that withdrawals transient storage var has increased by the borrowed amount.

Reply:
added